packages/passport-interface/src/api/requestChangeBlobKey.ts

```diff
@@ -22,20 +22,50 @@ export async function requestChangeBlobKey(
   newBlobKey: string,
   uuid: string,
   newSalt: string,
-  encryptedStorage: EncryptedPacket
+  encryptedStorage: EncryptedPacket,
+  baseRevision?: string
 ): Promise<ChangeBlobKeyResult> {
-  return httpPost(
+  return httpPost<ChangeBlobKeyResult>(
     urlJoin(zupassServerUrl, `/sync/changeBlobKey`),
     {
-      onValue: async () => ({ value: undefined, success: true }),
-      onError: async (resText) => JSON.parse(resText)
+      onValue: async (resText: string) => ({
+        value: JSON.parse(resText) as ChangeBlobKeyResponseValue,
+        success: true
+      }),
+      onError: async (resText: string) => {
+        // TODO(atwyman): Clean up this inconsistent client/server error handling pattern.
```

Comment:
what do you think should be done to clean it up?

Reply:
Working on it.